comictaggerlib/plugin_update_manager.py

```diff
@@ -0,0 +1,332 @@
+"""Manages installing/updating plugins"""
+
+# Copyright 2012-2014 ComicTagger Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import logging
+import os
+import sys
+import time
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, cast
+
+import yaml
+from packaging.version import InvalidVersion, Version, parse
+
+import comictaggerlib
+from comictaggerlib.ctsettings import ct_ns, plugin_finder
+
+try:
+    import niquests as requests
+except ImportError:
+    import requests
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class Plugin:
+    plugin_id: str
+    name: str
+    type: str
+    desc: str
+    manifest: str
+
+    @staticmethod
+    def from_yaml(yaml_data: list[dict[str, str]]) -> list[Plugin]:
+        return [Plugin(**entry) for entry in yaml_data]
+
+
+@dataclass
+class Download:
+    version: str
+    url: str
+
+
+@dataclass
+class PluginReleases:
+    latest: str
+    downloads: list[Download]
+
+    @staticmethod
+    def from_yaml(yaml_data: yaml.YAMLObject) -> PluginReleases:
+        downloads: list[Download] = [Download(**entry) for entry in yaml_data.get("downloads", [])]
+        return PluginReleases(latest=yaml_data["latest"], downloads=downloads)
+
+
+class PluginUpdateManager:
+    def __init__(self, config: ct_ns):
+        self.config = config
+        self.plugin_dir: Path = Path(self.config.Runtime_Options__config.user_plugin_dir)
+        self.plugin_files: dict[str, list[tuple[Version, Path]]] = {}
+        self.remote_plugin_list: list[Plugin] = []
+        self._read_plugin_list()
+        self._find_local_plugins()
+
+    def _parse_version(self, file: str) -> Version | None:
+        try:
+            return parse(file)
+        except InvalidVersion:
+            logger.error(f"Invalid version number for : {file}")
+            return None
+
+    def _find_local_plugins(self) -> None:
+        local_plugins = plugin_finder.find_plugins(self.plugin_dir)
+
+        # Create a deduped dict with each sorted by Version number
```

Comment:
All loading code should probably be handled in `plugin_finder.py` otherwise there can be inconsistencies

Reply:
I'm not understanding this as it's using `plugin_finder`. 

After that it just extracts the `entry_name`, `path` and `version`.